### PR TITLE
add searchOperator

### PR DESCRIPTION
**Pull request by shusharin**
Adding search operator for case insensitive search (Postgresql).
[Issue 110](https://github.com/laravel-enso/VueDatatable/issues/110)

### CODE REVIEW

**Comment by aocneanu** [comment]
Please take a look at how we send template options (buttons, meta) towards the frontend and then use them in the request. The same will apply for this case.

Lets use `comparisonOperator` instead of `searchOperator`

**Comment by shusharin** [comment]
Ok, i will fix it.

**Comment by aocneanu** [comment]
@shusharin ?

**Comment by shusharin** [comment]
Will do this on this week.

**Comment by shusharin** [comment]
@aocneanu hi! If all right, i can try prepare pull request in documentation repository.

**Comment by aocneanu** [comment]
Hi @shusharin 

>Please take a look at how we send template options (buttons, meta) towards the frontend and then use them in the request. The same will apply for this case.

We don't want to read the template again for the builder. We will just have to pass the new option on the `init` call and use it for the following fetches. Take a look how the package deals with other similar options.

**Comment by shusharin** [comment]
Yeah, i understood that. It was fixed in [last commit ](https://github.com/laravel-enso/VueDatatable/pull/111/commits/cea25bab11351921980dac2a6a7ba453d9149e64)
image

image

File: src/app/Classes/Table.php [hunk 1]
```diff
@@ -22,10 +22,14 @@ public function request()
...
+    /**
+     * init VueDatatable.
+     *
+     * @return array
+     */
     public function init()
     {
-        return (new Template($this->templatePath()))
-            ->get();
+        return ['template' => $this->getTemplate($this->templatePath())];
```
Comment (aocneanu) [inline src/app/Classes/Table.php]: good catch 👍

File: src/app/Classes/Template.php [hunk 2]
```diff
@@ -16,14 +16,19 @@ public function __construct(string $filename)
...
+     */
     public function get()
     {
         $this->readTemplate();
 
         (new Builder($this->template))
             ->run();
 
-        return ['template' => $this->template];
+        return $this->template;
```
Comment (aocneanu) [inline src/app/Classes/Template.php]: 👍

File: src/app/Classes/Table.php [hunk 3]
```diff
@@ -53,9 +57,22 @@ public function templatePath()
 
     private function builder()
     {
-        return new Builder(
-            $this->request,
-            $this->query()
-        );
+        // Call query before builder
+        // Sometimes we need declare template path in current request
+        $query = $this->query();
```
Comment (aocneanu) [inline src/app/Classes/Table.php]: It's not clear to me what you're doing here

File: src/app/Classes/Table.php [hunk 4]
```diff
@@ -53,9 +57,22 @@ public function templatePath()
...
+
+        return new Builder($this->request, $query, $this->getTemplate($this->templatePath()));
```
Comment (aocneanu) [inline src/app/Classes/Table.php]: We don't need the third argument (this is why I though you didn't refactor)

File: src/app/Classes/Table.php [hunk 5]
```diff
@@ -53,9 +57,22 @@ public function templatePath()
...
+    }
+
+    /**
+     * Get template object.
+     *
+     * @param string $path - path to template.json
+     *
+     * @return \stdClass
+     */
+    protected function getTemplate($path)
```
Comment (aocneanu) [inline src/app/Classes/Table.php]: What's this for?

File: src/config/datatable.php [hunk 6]
```diff
@@ -214,4 +214,17 @@
     */
 
     'dataRouteSuffix' => 'tableData',
+
+    /*
+    |--------------------------------------------------------------------------
+    | SQL comparison operator
+    |--------------------------------------------------------------------------
+    |
+    | The comparison operator will be the default used for every table. You will still
```
Comment (aocneanu) [inline src/config/datatable.php]: make this line shorter (move to the following line) to fit the "skeleton"

File: src/resources/js/components/enso/vuedatatable/VueTable.vue [hunk 7]
```diff
@@ -355,6 +355,7 @@ export default {
                 intervals: this.intervals,
                 params: this.params,
                 selected: this.selected,
+                comparisonOperator: this.template.comparisonOperator,
```
Comment (aocneanu) [inline src/resources/js/components/enso/vuedatatable/VueTable.vue]: this need to be added to the `exportRequest()` method also. We want in the excel export the same data as we get in the table.

File: src/app/Classes/Table.php (hunk 5 above)
Comment (shusharin) [reply src/app/Classes/Table.php]: Yeah, you are right. It's too much. Will fix it.

**Pull request merged**